internal/keystore/signer.go

```diff
@@ -26,7 +26,7 @@ import (
 func Signer(protocol string, pk crypto.PrivateKey) (signer.Signer, error) {
 	switch protocol {
 	case protocols.Ethereum:
-		return ethereum.NewSigner(pk), nil
+		return ethereum.NewSigner(pk)
```

Comment:
The changes in this file look out of scope for #350.
they appear related to #331 could you move them in to a seperate PR for that to make it clearer. That would be awesome as then its 2 PR's!!!!! 😄

Reply:
Thanks for the feedback!
I'm a little bit confused with issues #350  and #331, they appear same to me, with the only difference being that in the description of #350 that the public key type needs to be `ed25519 ` and in #331 `secp256k1`
Also the tile of #350 says to `ensure valid keys are used for nacl.Decrypter` isn't that already done [here](https://github.com/mailchain/mailchain/blob/master/crypto/cipher/nacl/decrypter.go#L12) ?

Reply:
Sorry for the lack of clarity @Despire let me add a bit more information.

- #350 is related encryption which is used to encrypt / decrypt messages this change is in the `nacl` package.
- #331 is related to the signing transactions, this change is in the `ethereum` package. 

In the case of signing a transaction and decrypting a message the private key is required. The code that you are implementing is very similar with the main difference being using `secp256k1` or `ed25519`.

No here is the part that's made it very confusing for which I am very sorry for! You are absolutely correct the [link](https://github.com/mailchain/mailchain/blob/master/crypto/cipher/nacl/decrypter.go#L12) does make that issue redundant. I will instead open an issue for `nacl.NewEncrypter()` #359

Reply:
Again thanks for the clarification 😄 
I guess this PR can be closed along with issue #350.